application/client.js

```diff
@@ -152,7 +152,14 @@ module.exports = HttpGateway.extend({
                     try {
                         responseData = JSON.parse(responseText);
                     } catch (e) {
-                        responseData = responseText;
```

Comment:
I guess we could do our own OPTIONS request.  The "real" request would hit it again though.  Thoughts, @paulstatezny ?